pkg/server/server.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package server
+
+import (
+	"context"
+	"net"
+
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/identity"
+)
+
+// Service represents a specific gRPC method collection to be registered
+// on a shared gRPC server. PointerDB, OverlayCache, PieceStore, Kademlia,
+// StatDB, etc. are all examples of services.
+type Service interface {
+	Run(ctx context.Context, server *Server) error
+}
+
+// Server represents a bundle of services defined by a specific ID.
+// Examples of servers are the satellite, the storagenode, and the uplink.
+type Server struct {
+	lis      net.Listener
+	grpc     *grpc.Server
+	next     []Service
```

Comment:
This would be removed eventually. The `Server` shouldn't start things, it complicates the whole bootstrap. But, probably a good transition. We can keep these as `Responsibility` to avoid confusion.